base/memory/ref_counted.h

```diff
@@ -318,6 +328,36 @@ class BASE_EXPORT ScopedAllowCrossThreadRefCountAccess final {
   static constexpr ::base::subtle::StartRefCountFromOneTag \
       kRefCountPreference = ::base::subtle::kStartRefCountFromOneTag
 
+// Use this macro to Assert ref count events of a ref counted class.
+// When using this macro, best also make sure it has a RecordReplayId().
+// NOTE: We use this messy macro magic so we only compile Replay diagnostics
+// for types that explicitely request it. This is better for performance
+// for uninstrumented types and avoids build errors in places where Replay is
+// not available.
+#define REPLAY_ASSERT_REF_COUNTS(className, label, assertWhenEventsDisallowed) \
+  static void replay_assert_ref_count_event(                                   \
+      const char* functionName, int refCount, int record_replay_id) {          \
+    if (recordreplay::IsRecordingOrReplaying() && label != 0) {                \
+      if (assertWhenEventsDisallowed || !recordreplay::AreEventsDisallowed())  \
+        recordreplay::Assert("[%s] %s::%s %d %d", label, className,            \
+                             functionName, record_replay_id, refCount);        \
+    }                                                                          \
+  }                                                                            \
+  static_assert(true, "require semicolon")
```

Comment:
What is the value of assertWhenEventsDisallowed ?  It is false in all the uses here.  Also, the label is always non-null, why do we test for it (and compare it with zero which is confusing).  All the branching here seems like it could be reduced to a single AssertMaybeEventsDisallowed call, which will be clearer and minimize overhead.